docs/sources/mimir/configure/configure-spread-minimizing-tokens/_index.md

```diff
@@ -0,0 +1,108 @@
+
+## Before you begin
+
+The ingester time series replication should be [configured with enabled zone-awareness](https://grafana.com/docs/mimir/latest/configure/configure-zone-aware-replication/#configuring-ingester-time-series-replication).
+
+{{% admonition type="note" %}}Use spread-mimizing tokens are recommended if [shuffle sharding](https://grafana.com/docs/mimir/latest/configure/configure-shuffle-sharding/#ingesters-shuffle-sharding) is disabled in your ingesters, or, if shuffle-sharding is enabled, but most of the tenants of your system use all available ingesters.
+{{% /admonition %}}
+
+For simplicity, let’s assume that there are three configured availability zones named `zone-a`, `zone-b`, and `zone-c`. Migration to the _spread-minimizing token generation strategy_ is a complex process performed zone by zone to prevent any data loss.
+
+## Step 1: Disable write requests to ingesters from zone-a
+
+To disable write requests, configure the following flag on the distributor and the ruler:
+
+```
+-ingester.ring.excluded-zones=zone-a
+```
+
+Before proceeding to the next step, use the following query to ensure that there are no write requests to the ingesters from `zone-a`:
+
+```
+sum by(route) (
+  rate(
+    cortex_request_duration_seconds_count{
+      namespace="<your-namespace>",
+      container="ingester",
+      pod=~"ingester-zone-a-.*",
+      route="/cortex.Ingester/Push"}[5m]
+  )
+)
+```
+
+You should see something like this:
+
+![No More Write Requests](no-more-write-requests.png)
+
+## Step 2: Shut down ingesters from zone-a
+
+Next, ensure that all the in-memory series of all ingesters from `zone-a` have been flushed to long-term storage, as well as that the ingesters from `zone-a` have been forgotten from the ring.
+To do this, invoke the [/ingester/shutdown](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/references/http-api/index.md#shutdown) endpoint on all the ingesters from `zone-a`.
+
+Before proceeding to the next step, ensure that all the calls completed successfully completed.
+
+## Step 3: Enable spread-minimizing tokens for ingesters in zone-a
+
+Configure the following flags on the ingesters from `zone-a`:
+
+```
+-ingester.ring.tokens-file-path=
+-ingester.ring.token-generation-strategy=spread-minimizing
+-ingester.ring.spread-minimizing-zones=zone-a,zone-b,zone-c
+```
+
+{{% admonition type="note" %}}
+The example uses `zone-a,zone-b,zone-c` to denote a comma-separated list of configured availability zones.
+{{% /admonition %}}
+
+Before proceeding to the next step, ensure that all the ingester pods related to `zone-a` are up and running with the new configuration.
+
+### Optional step: In-order registration of ingesters
+
+Mimir can force the ring to perform an in-order registration of ingesters, for example, to allow an ingester to register its tokens within the ring only after all previous ingesters (with ID lower than its own ID) have already been registered.
```

Comment:
Lines 74-76 had some run-on sentences with some complicated ideas. I tried to split them into several sentences. Does this look okay?